### PR TITLE
fix: enforce lowercase on AtKey

### PR DESCRIPTION
**Pull request by srieteja**
**- What I did**
- made key and namespace in AtKey class private variables and introduced getters and setters for the same
- enforcing toLowerCase() for all AtKey params in their respective setters
- set isHidden param of MetaData to true in PrivateKey class
- added isHidden check in AtKey.toString() to construct private keys when isHidden is set to true

**- How to verify it**
- test cases added to test/at_key_test.dart

**- Description for the changelog**

- chore: AtKey.key and AtKey.namespace made private(get and set introduced) and enforcing toLowerCase() for all AtKey params in their respective setters

### CODE REVIEW

**Comment by srieteja** [comment]
This pr is related to https://github.com/atsign-foundation/at_server/issues/964

File: packages/at_commons/lib/src/keystore/at_key.dart [hunk 1]
```diff
@@ -38,22 +38,35 @@ class AtKey {
   /// These keys will never be synced between the client and secondary server.
   bool _isLocal = false;
 
+  String? get key => _key;
+
+  set key(String? key) {
+    _key = key?.toLowerCase();
```
Comment (murali-shris) [inline packages/at_commons/lib/src/keystore/at_key.dart]: set to lower case only if passed key is not null or better make _key not null.

File: packages/at_commons/lib/src/keystore/at_key.dart [hunk 2]
```diff
@@ -38,22 +38,35 @@ class AtKey {
...
+  }
+
+  String? get namespace => _namespace;
+
+  set namespace(String? namespace) {
+    _namespace = namespace?.toLowerCase();
```
Comment (murali-shris) [inline packages/at_commons/lib/src/keystore/at_key.dart]: set to lower case only if namespace is not null

File: packages/at_commons/lib/src/keystore/at_key.dart (hunk 1 above)
Comment (srieteja) [reply packages/at_commons/lib/src/keystore/at_key.dart]: addressed as part of latest commit

File: packages/at_commons/lib/src/keystore/at_key.dart (hunk 2 above)
Comment (srieteja) [reply packages/at_commons/lib/src/keystore/at_key.dart]: addressed as part of the latest commit

**Review by murali-shris — APPROVED** [review]

**Pull request merged**